js/src/api/contract/contract.js

```diff
@@ -16,7 +16,9 @@
 
 import Abi from '../../abi';
 import Api from '../api';
-import { isInstanceOf } from '../util/types';
+import { isInstanceOf, isObject } from '../util/types';
+
+const DEFAULT_DEPLOY_GAS = 900000;
```

Comment:
Estimation is place. A small premium is added, which also allows checking that gasUsed == gas (all gas consumed) is as a result of a contract deployment exception, exception, not deployed.